localstack-core/localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -1270,6 +1285,8 @@ def apply_change(self, change: ChangeConfig, stack: Stack) -> None:
                 resource_provider, resource, resource_provider_payload
             )
         else:
+            resource["PhysicalResourceId"] = MOCK_REFERENCE
+            resource["Ref"] = MOCK_REFERENCE
```

Comment:
Why do we need to set the Ref here? :thinking:  Is this somehow overriding the physical resource id lookup if set?

Reply:
After some triage I found that the ref is not necessary, only the PhysicalResourceId is. And the assignment is after the search for the provider returns none. In the code you sent me it appeared that the assigment was inside the `try_load_resource_provider` method but this is not possible due to the function only receiving the resource type string.